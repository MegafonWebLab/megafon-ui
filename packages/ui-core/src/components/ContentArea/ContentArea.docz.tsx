export const wrapperStyles = {
    backgroundColor: '#E5F5EE',
    padding: '16px 0',
};
