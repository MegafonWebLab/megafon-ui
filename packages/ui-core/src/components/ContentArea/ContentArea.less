@import '~styles/base';

.mfui-beta-content-area {
    // mod: color

    &_color_white {
        background-color: @clearWhite;
    }

    &_color_transparent {
        background-color: transparent;
    }

    &_color_green {
        background-color: @green;
    }

    &_color_purple {
        background-color: @purple;
    }

    &_color_spbSky0 {
        background-color: @spbSky0;
    }

    &_color_spbSky1 {
        background-color: @spbSky1;
    }

    &_color_spbSky2 {
        background-color: @spbSky2;
    }

    &_color_freshAsphalt {
        background-color: @freshAsphalt;
    }

    &_color_fullBlack {
        background-color: @fullBlack;
    }

    &__inner {
        margin: 0 auto;
        max-width: 1440px;
        box-sizing: content-box;
        padding-left: @contentHorizontalIndentDesktopMU;
        padding-right: @contentHorizontalIndentDesktopMU;

        @media @mobileSM {
            padding-left: @contentHorizontalIndentMobileSM;
            padding-right: @contentHorizontalIndentMobileSM;
        }

        @media @mobileB {
            padding-left: @contentHorizontalIndentMobileB;
            padding-right: @contentHorizontalIndentMobileB;
        }

        @media @desktopS {
            padding-left: @contentHorizontalIndentDesktopS;
            padding-right: @contentHorizontalIndentDesktopS;
        }
    }

    // mod: disable-indents

    &__inner_disable-indents_mobile {
        @media @mobileSM {
            padding-left: 0;
            padding-right: 0;
        }
    }

    &__inner_disable-indents_mobile-tablet {
        @media @mobile {
            padding-left: 0;
            padding-right: 0;
        }
    }

    &__inner_disable-indents_tablet-desktop {
        @media @mobileBU {
            padding-left: 0;
            padding-right: 0;
        }
    }

    &__inner_disable-indents_all {
        padding-left: 0;
        padding-right: 0;
    }

    // mod: color

    &__inner_color_white {
        background-color: @clearWhite;
    }

    &__inner_color_transparent {
        background-color: transparent;
    }

    &__inner_color_green {
        background-color: @green;
    }

    &__inner_color_purple {
        background-color: @purple;
    }

    &__inner_color_spbSky0 {
        background-color: @spbSky0;
    }

    &__inner_color_spbSky1 {
        background-color: @spbSky1;
    }

    &__inner_color_spbSky2 {
        background-color: @spbSky2;
    }

    &__inner_color_freshAsphalt {
        background-color: @freshAsphalt;
    }

    &__inner_color_fullBlack {
        background-color: @fullBlack;
    }
}
