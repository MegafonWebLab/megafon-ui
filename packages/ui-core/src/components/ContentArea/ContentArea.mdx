---
name: ContentArea
menu: Компоненты
route: components/content-area
---

import { Playground, Props } from 'docz';
import ContentArea from './ContentArea';
import { wrapperStyles } from './ContentArea.docz';

# ContentArea

## Properties

<Props of={ContentArea} />

## Basic usage

<Playground>
    <ContentArea>Some content</ContentArea>
</Playground>

## Mobile inner padding

<Playground>
    <ContentArea outerBackgroundColor="spbSky0" mobileInnerPadding="none">
        It disable paddings for internal block on viewport width smaller than 768px.
        This can be seen on screens width less than 768 pixels.
    </ContentArea>
</Playground>

## None-padding usage

<Playground>
    <ContentArea innerPadding="none" innerBackgroundColor="white">
        Some content
    </ContentArea>
</Playground>

## You can choose the background color for an external or internal block

<Playground>
    <div style={wrapperStyles}>
        <ContentArea innerBackgroundColor="green">
            green
        </ContentArea>
    </div>

    <div style={wrapperStyles}>
        <ContentArea innerBackgroundColor="purple">
            <div style={{color: 'white'}}>purple</div>
        </ContentArea>
    </div>

        <div style={wrapperStyles}>
        <ContentArea innerBackgroundColor="transparent">
            transparent
        </ContentArea>
    </div>

    <div style={wrapperStyles}>
        <ContentArea innerBackgroundColor="white">
            white
        </ContentArea>
    </div>

    <div style={wrapperStyles}>
        <ContentArea innerBackgroundColor="spbSky0">
            spbSky0
        </ContentArea>
    </div>

    <div style={wrapperStyles}>
        <ContentArea innerBackgroundColor="spbSky1">
            spbSky1
        </ContentArea>
    </div>

    <div style={wrapperStyles}>
        <ContentArea innerBackgroundColor="spbSky2">
            spbSky2
        </ContentArea>
    </div>

    <div style={wrapperStyles}>
        <ContentArea innerBackgroundColor="freshAsphalt">
            <div style={{color: 'white'}}>freshAsphalt</div>
        </ContentArea>
    </div>

    <div style={{ ...wrapperStyles, marginBottom: 0 }}>
        <ContentArea innerBackgroundColor="fullBlack">
            <div style={{color: 'white'}}>fullBlack</div>
        </ContentArea>
    </div>
</Playground>
