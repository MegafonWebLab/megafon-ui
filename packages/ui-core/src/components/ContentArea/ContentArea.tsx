import * as React from 'react';
import * as PropTypes from 'prop-types';
import './ContentArea.less';
import cnCreate from 'utils/cnCreate';

export interface IProps {
    /** Background color of the external container */
    outerBackgroundColor?: 'white' | 'transparent';
    /** Background color of the internal container */
    innerBackgroundColor?: 'white' | 'transparent';
    /** Side paddings of the internal container */
    innerPadding?: 'lg' | 'none';
    /** Side paddings of the internal container on mobile screen */
    mobileInnerPadding?: 'default' | 'none';
    children: any;
    className?: string;
}

const cn = cnCreate('mfui-content-area');
class ContentArea extends React.Component<IProps, {}> {
    static propTypes = {
        outerBackgroundColor: PropTypes.oneOf(['white', 'transparent']),
        innerBackgroundColor: PropTypes.oneOf(['white', 'transparent']),
        innerPadding: PropTypes.oneOf(['lg', 'none']),
        mobileInnerPadding: PropTypes.oneOf(['default', 'none']),
        children: PropTypes.node,
        className: PropTypes.string,
    };

    static defaultProps = {
        outerBackgroundColor: 'transparent',
        innerBackgroundColor: 'transparent',
        innerPadding: 'lg',
        mobileInnerPadding: 'default',
    };

    render() {
        const {
            outerBackgroundColor,
            innerBackgroundColor,
            innerPadding,
            mobileInnerPadding,
            children,
            className,
        } = this.props;

        return (
            <div className={cn('', { color: outerBackgroundColor }, className)}>
                <div
                    className={cn('inner', {
                        padding: innerPadding,
                        'mobile-padding': mobileInnerPadding,
                        color: innerBackgroundColor,
                    })}
                >
                    {children}
                </div>
            </div>
        );
    }
}

export default ContentArea;
