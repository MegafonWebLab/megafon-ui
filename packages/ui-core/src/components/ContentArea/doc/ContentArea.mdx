---
name: ContentArea
menu: Компоненты
route: components/content-area
---

import { Props } from 'docz';
import ContentArea from '../ContentArea';
import Example from './ContentArea.example.mdx';

# ContentArea

Компонент задает содержимому страницы боковые отступы,
максимальную ширину и центрирование на больших разрешениях

<CopyToClipboardBox text="import { ContentArea } from '@megafon/ui-core';" />

<DoczTabs>
    <Example />
    <Props of={ContentArea} />
</DoczTabs>
