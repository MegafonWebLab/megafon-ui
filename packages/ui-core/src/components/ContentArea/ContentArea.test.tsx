import * as React from 'react';
import { shallow } from 'enzyme';
import ContentArea, { IProps, BackgroundColorType } from './ContentArea';

const props: Partial<IProps> = {
    outerBackgroundColor: 'white',
    innerBackgroundColor: 'white',
    innerPadding: 'none',
    mobileInnerPadding: 'none',
    className: 'test-class',
};

const backgroundColors = [
    'white',
    'transparent',
    'green',
    'purple',
    'spbSky0',
    'spbSky1',
    'spbSky2',
    'soggyAsphalt',
    'fullBlack',
];

describe('<ContentArea />', () => {
    const getWrapper = (additionalProps?: Partial<IProps>) =>
        shallow(
            <ContentArea {...additionalProps}>
                <span>child</span>
            </ContentArea>
        );

    it('renders correctly with default props', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with custom props', () => {
        const wrapper = getWrapper(props);
        expect(wrapper).toMatchSnapshot();
    });

    backgroundColors.forEach(color => {
        it(`render component when outerBackgroundColor with ${color}`, () => {
            const wrapper = shallow(
                <ContentArea
                    outerBackgroundColor={color as BackgroundColorType}
                >
                    some content
                </ContentArea>
            );
            expect(wrapper).toMatchSnapshot();
        });
    });

    backgroundColors.forEach(color => {
        it(`render component when innerBackgroundColor with ${color}`, () => {
            const wrapper = shallow(
                <ContentArea
                    innerBackgroundColor={color as BackgroundColorType}
                >
                    some content
                </ContentArea>
            );
            expect(wrapper).toMatchSnapshot();
        });
    });
});
