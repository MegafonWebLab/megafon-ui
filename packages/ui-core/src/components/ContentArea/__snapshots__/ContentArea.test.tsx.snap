// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentArea /> render component when innerBackgroundColor with fullBlack 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_fullBlack"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when innerBackgroundColor with green 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_green"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when innerBackgroundColor with purple 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_purple"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when innerBackgroundColor with soggyAsphalt 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_soggyAsphalt"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when innerBackgroundColor with spbSky0 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_spbSky0"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when innerBackgroundColor with spbSky1 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_spbSky1"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when innerBackgroundColor with spbSky2 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_spbSky2"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when innerBackgroundColor with transparent 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when innerBackgroundColor with white 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_white"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when outerBackgroundColor with fullBlack 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_fullBlack"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when outerBackgroundColor with green 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_green"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when outerBackgroundColor with purple 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_purple"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when outerBackgroundColor with soggyAsphalt 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_soggyAsphalt"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when outerBackgroundColor with spbSky0 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_spbSky0"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when outerBackgroundColor with spbSky1 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_spbSky1"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when outerBackgroundColor with spbSky2 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_spbSky2"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when outerBackgroundColor with transparent 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> render component when outerBackgroundColor with white 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_white"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    some content
  </div>
</div>
`;

exports[`<ContentArea /> renders correctly with custom props 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_white test-class"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_none mfui-content-area__inner_mobile-padding_none mfui-content-area__inner_color_white"
  >
    <span>
      child
    </span>
  </div>
</div>
`;

exports[`<ContentArea /> renders correctly with default props 1`] = `
<div
  className="mfui-content-area mfui-content-area_color_transparent"
>
  <div
    className="mfui-content-area__inner mfui-content-area__inner_padding_default mfui-content-area__inner_mobile-padding_default mfui-content-area__inner_color_transparent"
  >
    <span>
      child
    </span>
  </div>
</div>
`;
